src/main.rs

```diff
@@ -224,7 +225,10 @@ async fn main() -> Result<(), Box<dyn Error>> {
     let keypair = app_config.consensus.keypair().clone();
 
     let (system_tx, mut system_rx) = mpsc::channel::<SystemMessage>(100);
-    let (mempool_tx, mempool_rx) = mpsc::channel(app_config.mempool.queue_size as usize);
+    let (user_message_mempool_tx, user_message_mempool_rx) =
+        mpsc::channel(app_config.mempool.queue_size as usize);
+    let (system_message_mempool_tx, system_message_mempool_rx) =
+        mpsc::channel(app_config.mempool.queue_size as usize);
```

Comment:
Why do we need two separate channels? So we don't drop messages if it's full?

Reply:
Yeah, this is more relevant when the mempool size is limited. When that's the case, I didn't want an overload of user messages to result in onchain events being dropped or stalled.

Reply:
When the mempool size is unlimited, the only thing that matters is the order we read out of the mempool in.